scalding-core/src/main/java/com/twitter/scalding/cascading/HashJoinDistCacheTransformer.java

```diff
@@ -0,0 +1,122 @@
+package com.twitter.scalding;
+
+import java.util.Set;
+
+import cascading.flow.FlowElement;
+import cascading.flow.planner.iso.expression.ElementCapture;
+import cascading.flow.planner.iso.expression.ExpressionGraph;
+import cascading.flow.planner.iso.expression.FlowElementExpression;
+import cascading.flow.planner.iso.expression.PathScopeExpression;
+import cascading.flow.planner.iso.finder.Match;
+import cascading.flow.planner.iso.transformer.ElementFactory;
+import cascading.flow.planner.iso.transformer.GraphTransformer;
+import cascading.flow.planner.iso.transformer.ReplaceGraphTransformer;
+import cascading.flow.planner.iso.transformer.Transformed;
+import cascading.flow.planner.graph.ElementGraph;
+import cascading.flow.planner.graph.ElementGraphs;
+import cascading.flow.planner.rule.PlanPhase;
+import cascading.flow.planner.rule.RuleExpression;
+import cascading.flow.planner.rule.RuleTransformer;
+import cascading.flow.planner.rule.expressiongraph.SyncPipeExpressionGraph;
+import cascading.pipe.HashJoin;
+import cascading.tap.hadoop.Hfs;
+import cascading.tap.hadoop.util.TempHfs;
+
+import static cascading.flow.planner.rule.PlanPhase.BalanceAssembly;
+
+class HashJoinExpression extends RuleExpression {
+
+  public HashJoinExpression() {
+    super(
+      new SyncPipeExpressionGraph(),
+      new ExpressionGraph()
+        .arc(
+          new FlowElementExpression(ElementCapture.Primary, Hfs.class),
+          PathScopeExpression.BLOCKING,
+          new FlowElementExpression(HashJoin.class)
+        )
```

Comment:
yeah, that should do it. Now the factory doesn't have to do so much work. But make this the last rule in the phase.